app/workers/delete_object_hierarchy_worker.rb

```diff
@@ -73,49 +74,71 @@ def batch_callbacks(batch)
     end
   end
 
-  def delete_associations
-    destroyable_associations.each do |association|
-      ReflectionDestroyer.new(object, association, caller_worker_hierarchy).destroy_later
+  def destroy_and_delete_associations
+    Array(object.background_deletion).each do |association, options|
```

Comment:
If you do it like https://github.com/3scale/porta/pull/1372/files#r348064637, then this would be just `object.background_deletion.each do |association, options|`